server/etcdserver/server.go

```diff
@@ -1202,7 +1202,7 @@ func (s *EtcdServer) MoveLeader(ctx context.Context, lead, transferee uint64) er
 	return nil
 }
 
-// TransferLeadership transfers the leader to the chosen transferee.
+// TransferLeadership transfers the leader to the chosen transferee. It is only used in server graceful shutdown.
```

Comment:
This is a good addition to comment that shows that function is badly named. This function should not be called outside of graceful shutdown. I would not call it leader transfer as it does more then `MoveLeader`.

I would suggest renaming it to something like `PassOnLeaderIfPossible`, `pass on` implies that this member will not be present for long, and `if possible` alludes to this function will not fail if there is no successor.

Reply:
Agreed. 

How about `tryLeaderTransferOnStop`?

Reply:
`tryLeaderTransferOnShutdown`?